format-translators-module/format-translators-module/agent/structure-translator-agent/structure_translator_agent.cpp

```diff
@@ -0,0 +1,88 @@
+#include "structure_translator_agent.hpp"
+
+#include "constants/format_translators_constants.hpp"
+#include "keynodes/format_translators_keynodes.hpp"
+
+namespace formatTranslators
+{
+ScAddr StructureTranslatorAgent::GetEventSubscriptionElement() const
+{
+  return FormatTranslatorsKeynodes::ui_command_initiated;
+}
+
+ScAddr StructureTranslatorAgent::GetActionClass() const
+{
+  return FormatTranslatorsKeynodes::ui_command_translate_from_sc;
+}
+
+bool StructureTranslatorAgent::CheckInitiationCondition(ScActionInitiatedEvent const & event)
+{
+  ScAddr const & command = event.GetOtherElement();
+  if (!m_context.CheckConnector(GetActionClass(), command, ScType::ConstPermPosArc))
+    return false;
+  auto const & formatIterator = m_context.CreateIterator5(
+      command,
+      ScType::ConstPermPosArc,
+      FormatTranslatorsKeynodes::format_scg_json,
+      ScType::ConstPermPosArc,
+      FormatTranslatorsKeynodes::ui_rrel_output_format);
+  if (!formatIterator->Next())
+    return false;
+  auto const & structureToTranslateIterator = m_context.CreateIterator5(
+      command,
+      ScType::ConstPermPosArc,
+      ScType::ConstNode,
+      ScType::ConstPermPosArc,
+      FormatTranslatorsKeynodes::ui_rrel_source_sc_construction);
+  if (!structureToTranslateIterator->Next())
+    return false;
+  structureToTranslate = structureToTranslateIterator->Get(2);
+  return m_context.IsElement(structureToTranslate);
+}
+
+ScResult StructureTranslatorAgent::DoProgram(
+    ScEventAfterGenerateOutgoingArc<ScType::ConstPermPosArc> const & event,
+    ScAction & executedAction)
+{
+  try
```

Comment:
Not all code should be wrapped in try/catch statement

Reply:
current solution gives more control to consumer